search_gov_crawler/dap/datastore.py

```diff
@@ -0,0 +1,64 @@
+import logging
+from datetime import UTC, datetime, timedelta
+
+from redis import Redis
+
+log = logging.getLogger(__name__)
+
+
+def write_dap_record_to_redis(redis: Redis, domain: str, date: str, visits: int) -> None:
+    """
+    Writes a DAP record to Redis.
+    """
+
+    key = f"dap_visits:{domain}"
+    name = str(visits)
+    score = int(date.replace("-", ""))
+
+    redis.zadd(key, {name: score})
+
+
+def age_off_dap_records(redis: Redis, days_back: int) -> int:
+    """
+    Removes DAP records older than `days_back` days.
+    """
+
+    dap_records_aged_off = 0
+    age_off_date = datetime.now(UTC) - timedelta(days=days_back)
+    min_score = int(age_off_date.strftime("%Y%m%d"))
+    keys = redis.keys("dap_visits:*")
```

Comment:
redis.keys can be an expensive (and blocking) operation (if we have lots of keys). Maybe you can do a scan instead?
```suggestion
    keys = list(redis.scan_iter("dap_visits:*"))
```